### PR TITLE
[#2610] feat(kafka-catalog, core): Adapt the Kafka catalog to CatalogOperationsDispatcher

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

This PR add the Kafka catalog adaption to the CatalogOperationsDispatcher

### Why are the changes needed?

Part of Kafka catalog support work

Fix: #2610 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

UTs

### CODE REVIEW

File: catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java [hunk 1]
```diff
@@ -204,11 +212,26 @@ public Topic createTopic(
...
+          topicId.toString(),
           createTopicsResult.numPartitions(ident.name()).get(),
           createTopicsResult.replicationFactor(ident.name()).get());
+
+      return KafkaTopic.builder()
+          .withName(ident.name())
+          .withComment(comment)
+          // Because there is no way to store the Gravitino ID in Kafka, therefor we use the topic
+          // ID as the Gravitino ID
+          .withProperties(newPropertiesWithId(convertToGravitinoId(topicId), properties))
```
Comment (jerryshao) [inline catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java:228]: Can you please add a test for this change.

File: core/src/main/java/com/datastrato/gravitino/catalog/TopicOperationDispatcher.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    // Case 1: The topic is not created by Gravitino.
+    // Note: for Kafka catalog, stringId will not be null. Because there is no way to store the
+    // Gravitino
+    // ID in Kafka, therefor we use the topic ID as the Gravitino ID
+    if (stringId == null) {
+      return EntityCombinedTopic.of(topic)
+          .withHiddenPropertiesSet(
+              getHiddenPropertyNames(
+                  catalogIdent, HasPropertyMetadata::topicPropertiesMetadata, topic.properties()));
+    }
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/catalog/TopicOperationDispatcher.java:88]: This should never happened, right?

File: core/src/main/java/com/datastrato/gravitino/storage/relational/JDBCBackend.java [hunk 3]
```diff
@@ -88,6 +88,7 @@ public <E extends Entity & HasIdentifier> void insert(E e, boolean overwritten)
     } else if (e instanceof FilesetEntity) {
       FilesetMetaService.getInstance().insertFileset((FilesetEntity) e, overwritten);
     } else {
+      // TODO: Add support for TopicEntity
       throw new UnsupportedEntityTypeException(
           "Unsupported entity type: %s for insert operation", e.getClass());
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/JDBCBackend.java:93]: Are you going to do this in another PR?

File: core/src/main/java/com/datastrato/gravitino/catalog/TopicOperationDispatcher.java (hunk 2 above)
Comment (mchades) [reply core/src/main/java/com/datastrato/gravitino/catalog/TopicOperationDispatcher.java:88]: As of now, yes. 

But if other messaging catalogs that support setting topic property are implemented later, such as Pulsar, this `stringId` may be null.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/JDBCBackend.java (hunk 3 above)
Comment (mchades) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/JDBCBackend.java:93]: Yes, some design work may be needed.

**Review by jerryshao — APPROVED** [review]
LGTM.
